src/extension/vscode/client.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { ApolloClient } from "@apollo/client";
+import { getPrivateAccess } from "../../privateAccess";
+import { createActor } from "../actor";
+import { createRpcClient, createRpcHandler } from "../rpc";
+import { getQueries, getMutations } from "../tab/helpers";
+import { loadErrorCodes } from "../tab/loadErrorCodes";
+import type { MessageAdapter } from "../messageAdapters";
+import { handleExplorerRequests } from "../tab/handleExplorerRequests";
+import { setMaxListeners, WeakRef, FinalizationRegistry } from "./polyfills";
+
+type Reason =
+  | "WS_DISCONNECTED"
+  | "CLIENT_GC"
+  | "WS_ERROR"
+  | "UNREGISTER"
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | (string & {});
+
+export function registerClient(
+  client: ApolloClient<any>,
+  url: string | URL
```

Comment:
> Should we have a default value here of ws://localhost:7095? Is it safe to assume that the majority of people will probably use the default VSCode configuration locally?

I would actually assume that for most people it would be a tunnel address - I wouldn't want to hide this with a default here.